command/agent/cache/api_proxy.go

```diff
@@ -36,7 +37,16 @@ func (ap *APIProxy) Send(ctx context.Context, req *SendRequest) (*SendResponse,
 		return nil, err
 	}
 	client.SetToken(req.Token)
-	client.SetHeaders(req.Request.Header)
+
+	// http.Transport will transparently request gzip and decompress the response, but only if
+	// the client doesn't manually set the header (in which case the client has to handle
+	// the decompression. If gzip has already been set, remove it to avoid triggering the manual
+	// handling requirement.
+	h := clone(req.Request.Header)
+	if h.Get("Accept-Encoding") == "gzip" {
```

Comment:
`h.Get` would only access the first value if multiple are provided so it would fail to check against something like `Accept-Encoding: compress, gzip`. We might what to access the map directly and ok check it instead.

Reply:
Yeah. Plus we'd only want to remove `gzip`, not other encodings. I'll update it.